docs/docs/reference/augments/method-augments.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: doc-page
+title: "Method Augmentations"
+---
+
+Method augmentations are a way to define _extension methods_. Here is a simple one:
+
+```scala
+case class Circle(x: Double, y: Double, radius: Double)
+
+augment Circle {
+  def circumference: Double = this.radius * math.Pi * 2
+}
+```
+
+The `augment Circle` clause adds an extension method `circumference` to values of class `Circle`. Like regular methods, extension methods can be invoked with infix `.`:
+
+```scala
+  val circle = Circle(0, 0, 1)
+  circle.circumference
+```
+
+### Meaning of `this`
+
+Inside an extension method, the name `this` stands for the receiver on which the
+method is applied when it is invoked. E.g. in the application of `circle.circumference`,
+the `this` in the body of `circumference` refers to `circle`. Unlike for regular methods,
+an explicit `this` is mandatory to refer to members of the receiver. So the following
```

Comment:
> Unlike for regular methods, an explicit `this` is mandatory to refer to members of the receiver.

Why? I can imagine implementation-dependent reasons, but it's harder to tell why it makes sense for the user.

Reply:
Whilst I understand the reasoning, it is also yet another local rule.